### DISCUSSION TITLE
Support direct paste file to file-dock

### DISCUSSION THREAD

**Discussion by jzq740176597** [godotengine/godot-proposals#6630] [Editor]
### Describe the project you are working on

3d

### Describe the problem or limitation you are having in your project

I have to select **target-folder** in file-dock => right-click open it in file-explorer -> paste file -> close & back to gd.
 **tedious**!

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

ctrl + V => direct paste file-asset to file-dock

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

*

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no

### Is there a reason why this should be core and not an add-on in the asset library?

core

**Comment by YuriSizov** [comment]
You can just drag'n'drop, by the way.

**Comment by timothyqiu** [comment]
I don't like the idea of pasting OS level objects into the editor.

Instead, I would like to see a proper "Import" button in the filesystem dock that let the user select files to copy. There was a pending PR for this, but I can't find it right now.

**Reply by Calinou** [reply]
> I don't like the idea of pasting OS level objects into the editor.

Out of curiosity, why not? There's been numerous times where I want something like this.

**Reply by timothyqiu** [reply]
Currently, there is no way for the user to actively request the editor to select and import some files. The solution is either to move the files outside the editor or drag them into the editor. Adding support for "pasting files directly" feels like another way to workaround the problem, which is not significantly different from dragging files into the editor.

I think it'll be priceless if it does something unique. For example: right click an image in the browser and select Copy Image. Pasting in the editor creates the image file out of nowhere.
